packages/core/test/persistence.spec.ts

```diff
@@ -0,0 +1,175 @@
+import { expect, describe, it, beforeEach, afterEach } from "vitest";
+import { getDefaultLoggerOptions, pino } from "@walletconnect/logger";
+import { ICore, IStore } from "@walletconnect/types";
+import { TEST_CORE_OPTIONS, disconnectSocket, throttle } from "./shared";
+import { Core, Store } from "../src";
+import { generateRandomBytes32 } from "@walletconnect/utils";
+
+let core: ICore;
+let store: IStore<any, any>;
+const n_restarts = 1; // number of restarts to use for persistence tests
+
+const logger = pino(getDefaultLoggerOptions({ level: "fatal" }));
+type MockValue = { id: string; value: string };
+const MOCK_STORE_NAME = "persistent-store";
+const storeTestValues = [
+  { id: "1", value: "foo" },
+  { id: "2", value: "bar" },
+  { id: "3", value: "baz" },
+];
+
+const initCore = async (dbName = "persistent-test") => {
+  const coreOptions = {
+    ...TEST_CORE_OPTIONS,
+    storageOptions: { database: `tmp/${dbName}.db` },
+  };
+  core = new Core(coreOptions);
+  await core.start();
+};
```

Comment:
```suggestion
const initCore = async (customOpts:  CoreTypes.Options) => {
  const coreOptions = {
    ...TEST_CORE_OPTIONS,
    ...customOpts
  };
  const core = new Core(coreOptions);
  await core.start();
  return core;
};
```
Could we make this utl fx dynamic so you could use it to create as many `Core` instances as needed? 
When calling later, you could specify the DB options
```
const core = initCore({ storageOptions: { database: "persistent-test" }})
```
It would allow us to expand the tests without having to rework the existing tests.

If a global variables are needed, instead of top-level `core`/`store` instances, we could scope them within the `describe` thus allowing further adjustments